src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+            $lockNotes = "This is in use by Azure Migrate project"
+            $lsaLock = Get-AzResourceLock -LockName $CommonLockName -ResourceName $LogStorageAcName -ResourceType $StorageType -ResourceGroupName $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$lsaLock){
+                New-AzResourceLock -LockLevel CanNotDelete -LockName $CommonLockName -ResourceName $LogStorageAcName -ResourceType $StorageType -ResourceGroupName $ResourceGroupName -LockNotes $lockNotes -Force
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $LogStorageAcName
+            }else{
+                Write-Host $CommonLockName, " for ", $LogStorageAcName, $LogStringSkipping
+            }
+            
+            $gwyLock = Get-AzResourceLock -LockName $CommonLockName -ResourceName $GateWayStorageAcName -ResourceType $StorageType -ResourceGroupName $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$gwyLock){
+                New-AzResourceLock -LockLevel CanNotDelete -LockName $CommonLockName -ResourceName $GateWayStorageAcName -ResourceType $StorageType -ResourceGroupName $ResourceGroupName -LockNotes $lockNotes -Force
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $GateWayStorageAcName
+            }else{
+                Write-Host $CommonLockName, " for ", $LogStorageAcName, $LogStringSkipping
+            }
+
+            $sbsnsLock = Get-AzResourceLock -LockName $CommonLockName -ResourceName $ServiceBusNamespace -ResourceType $ServiceBusType -ResourceGroupName $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$sbsnsLock){
+                New-AzResourceLock -LockLevel CanNotDelete -LockName $CommonLockName -ResourceName $ServiceBusNamespace -ResourceType $ServiceBusType -ResourceGroupName $ResourceGroupName -LockNotes $lockNotes -Force
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $ServiceBusNamespace
+            }else{
+                Write-Host $CommonLockName, " for ", $ServiceBusNamespace, $LogStringSkipping
+            }
+
+            $kvLock = Get-AzResourceLock -LockName $CommonLockName -ResourceName $KeyVaultName -ResourceType $KeyVaultType -ResourceGroupName $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$kvLock){
+                New-AzResourceLock -LockLevel CanNotDelete -LockName $CommonLockName -ResourceName $KeyVaultName -ResourceType $KeyVaultType -ResourceGroupName $ResourceGroupName -LockNotes $lockNotes -Force
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $KeyVaultName
+            }else{
+                Write-Host $CommonLockName, " for ", $KeyVaultName, $LogStringSkipping
+            }
+            
+
+            # Intermediate phase
+            # RoleAssignments
+            
+            $roleDefinitionId = "81a9662b-bebf-436f-a333-f67b29880f12"
```

Comment:
for understanding plz rename roleDefinitionId => storageOperatorRoleDefinitionId